ipld/cid/src/error.rs

```diff
@@ -4,10 +4,10 @@
 use encoding::Error as EncodingError;
 use multibase;
 use multihash;
+use serde::{Deserialize, Serialize};
 use std::{error, fmt, io};
-
 /// Error types
-#[derive(PartialEq, Eq, Clone, Debug)]
+#[derive(PartialEq, Eq, Clone, Debug, Serialize, Deserialize)]
```

Comment:
remove serialize and deserialize, you only have this because you are trying to serialize this in the amt